src/dioptra/restapi/v1/users/controller.py

```diff
@@ -0,0 +1,152 @@
+# license (CC BY 4.0). The disclaimers of the CC BY 4.0 license apply to all parts
+# of the software developed or licensed by NIST.
+#
+# ACCESS THE FULL CC BY 4.0 LICENSE HERE:
+# https://creativecommons.org/licenses/by/4.0/legalcode
+"""The module defining the endpoints for User resources."""
+from __future__ import annotations
+
+import uuid
+
+import structlog
+from flask import request
+from flask_accepts import accepts, responds
+from flask_login import login_required
+from flask_restx import Namespace, Resource
+from structlog.stdlib import BoundLogger
+
+from dioptra.restapi.v1.schemas import IdStatusResponseSchema
+
+from .schema import (
+    UserCurrentSchema,
+    UserGetQueryParameters,
+    UserMutableFieldsSchema,
+    UserPageSchema,
+    UserPasswordSchema,
+    UserSchema,
+)
+
+LOGGER: BoundLogger = structlog.stdlib.get_logger()
+
+api: Namespace = Namespace("Users", description="Users endpoint")
+
+
+@api.route("/")
+class UserEndpoint(Resource):
+
+    @login_required
+    @accepts(query_params_schema=UserGetQueryParameters, api=api)
+    @responds(schema=UserPageSchema, api=api)
+    def get(self):
+        """Gets a list of all Users."""
+        log = LOGGER.new(
+            request_id=str(uuid.uuid4()), resource="User", request_type="GET"
+        )
+        log.debug("Request received")
+        parsed_query_params = request.parsed_obj  # noqa: F841
+
+    @login_required
+    @accepts(schema=UserSchema, api=api)
+    @responds(schema=UserCurrentSchema, api=api)
+    def post(self):
+        """Creates a User."""
+        log = LOGGER.new(
+            request_id=str(uuid.uuid4()), resource="User", request_type="POST"
+        )
+        log.debug("Request received")
+        parsed_obj = request.parsed_obj  # noqa: F841
+
+
+@api.route("/<int:id>")
+@api.param("id", "ID for the User resource.")
+class UserIdEndpoint(Resource):
+
+    @login_required
+    @responds(schema=UserSchema, api=api)
+    def get(self, id: int):
+        """Gets the User with the provided ID."""
+        log = LOGGER.new(
+            request_id=str(uuid.uuid4()), resource="User", request_type="GET", id=id
+        )
+        log.debug("Request received")
+
+
+@api.route("/current")
+class UserCurrentEndpoint(Resource):
+
+    @login_required
+    @responds(schema=UserCurrentSchema, api=api)
+    def get(self):
+        """Gets the Current User."""
+        log = LOGGER.new(
+            request_id=str(uuid.uuid4()), resource="User", request_type="GET"
+        )
+        log.debug("Request received")
+
+    @login_required
+    @responds(schema=IdStatusResponseSchema, api=api)
+    def delete(self):
+        """Deletes a Current User."""
+        log = LOGGER.new(
+            request_id=str(uuid.uuid4()),
+            resource="User",
+            request_type="DELETE",
+        )
+        log.debug("Request received")
+
+    @login_required
+    @accepts(schema=UserMutableFieldsSchema, api=api)
+    @responds(schema=UserSchema, api=api)
```

Comment:
The current PUT should return the same fields that a /current GET provides.

```suggestion
    @responds(schema=UserCurrentSchema, api=api)
```